### PR TITLE
feat: add `ParOps.mapValues`

### LINKED ISSUE #2974: Add `ParOps.mapValues` for `Map[K, V]` and use it broadly
The idea is to have a function `mapValues` which applies a function `V1 => V2` to all values in a Map in parallel and rebuilds the map.

We can then use it in several places in the compiler.

### PR DESCRIPTION
**Pull request by sockmaster27**
Related #2974

### CODE REVIEW

**Comment by magnus-madsen** [comment]
We also need to figure out where to use it. A few tests might be nice too.

**Comment by magnus-madsen** [comment]
Going to ask @mlutze to comment.

File: main/src/ca/uwaterloo/flix/util/ParOps.scala [hunk 1]
```diff
@@ -43,6 +43,15 @@ object ParOps {
     result.seq
   }
 
+  /**
+    * Apply the given function `f` to each value in the map `m` in parallel.
+    */
+  @inline
+  def mapValues[K, A, B](m: Map[K, A])(f: A => B)(implicit flix: Flix): Map[K, B] =
+    parMap(m) {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/util/ParOps.scala:51]: A question is whether using the parMap function here creates too many intermediate data structures.

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/util/ParOps.scala:51]: Maybe there is a nice general interface that all 3 functions can use.

**Review by mlutze — APPROVED** [review]

**Comment by mlutze** [comment]
Can this be merged in so I can use it? :)

**Comment by mlutze** [comment]
Would also be nice to have some helpers for dealing with parallel processing things that return Validations

**Comment by magnus-madsen** [comment]
But I would like parValues to be used where parMap is used today (on maps), and I would like it to be more efficient.

**Comment by mlutze** [comment]
> But I would like parValues to be used where parMap is used today (on maps), and I would like it to be more efficient.

Can't that be done as a next step?

**Comment by magnus-madsen** [comment]
> > But I would like parValues to be used where parMap is used today (on maps), and I would like it to be more efficient.
> 
> Can't that be done as a next step?

Okay

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**